api/logs/v1/labels_enforcer.go

```diff
@@ -70,21 +71,24 @@ func WithEnforceAuthorizationLabels() func(http.Handler) http.Handler {
 func enforceValues(mInfo AuthzResponseData, u *url.URL) (values string, err error) {
 	switch {
 	case strings.HasSuffix(u.Path, "/values"):
-		return enforceValuesOnLabelValues(mInfo, u.Query())
+		return enforceValuesOnFilter(mInfo, u.Query(), queryParam)
+	case strings.HasSuffix(u.Path, "/series"):
+		return enforceValuesOnFilter(mInfo, u.Query(), matchParam)
 	default:
 		return enforceValuesOnQuery(mInfo, u.Query())
 	}
 }
 
-func enforceValuesOnLabelValues(mInfo AuthzResponseData, v url.Values) (values string, err error) {
+func enforceValuesOnFilter(mInfo AuthzResponseData, v url.Values, filterParam string) (values string, err error) {
```

Comment:
This function and `enforceValuesOnQuery` seem to be very very similar to me (maybe except for the handling of `or`). I wonder if it makes sense to give refactoring them to share more code a try, for example by splitting the "extraction" from the "building a matcher" parts.